poc_v2.1_revised/scraping_load.py

```
@@ -0,0 +1,247 @@
+        min_score=RERANK_MIN_SCORE,
+        top_k=max(k, MIN_LT_RESULTS + MIN_LAW_RESULTS),
+    )
+
+    # 4) Final balancing pass:
+    #    - for general: purely top-k after rerank
+    #    - for tenant/landlord: try to keep at least one role-matched LT doc if available
+    if role and role != "general":
+        role_docs = []
+        non_role_docs = []
+
+        for doc, score in zip(final_docs, final_scores):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)